src/DotNetWorker.Core/Converters/DateTimeConverter.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Functions.Worker.Converters
+{
+    /// <summary>
+    /// Converter to bind to DateTime/DateTimeOffset type.
+    /// </summary>
+    internal sealed class DateTimeConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            if (!IsValidTargetType(context) || context.Source is not string source)
+            {
+                return new ValueTask<ConversionResult>(ConversionResult.Unhandled());
+            }
+
+            if ((context.TargetType == typeof(DateTimeOffset) || context.TargetType == typeof(DateTimeOffset?))
```

Comment:
In what scenario would we pass `null` to a `DateTimeOffset?` or `DateTime?`? Is there any?

Reply:
Let's say customer has a function definition like below.

````
public static HttpResponseData Run(
    [HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequestData req,
    DateTime startDate,
    DateTime? endDate)
````

where endDate param is nullable type / optional.  The below 2 requests will work for this function definition. 

````
api/Function4?startdate=2021/10/10&enddate=2021/12/12
api/Function4?startdate=2021/12/12
````
For the second request where `endDate` is not part of the request's input data/ trigger metadata, so, we do not have a source data to use (ie: `source` is null). So, the converter will return **_Unhandled_** result and the default value of the [parameterValues array (of type `object[]`) ](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/src/DotNetWorker.Core/Context/Features/DefaultModelBindingFeature.cs#L15)entry will be used, which is `null`. We will end up passing [null as that argument's value when invoking the function](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/src/DotNetWorker.Core/Invocation/MethodInvokerWithReturnValue.cs#L22).

Reply:
We do similar thing for [Guid converter where `Guid?` is supported](url).